cmd/record/record.go

```diff
@@ -0,0 +1,217 @@
+			filters, err := parseFilters(args)
+			if err != nil {
+				return err
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			hubbleConn, err := conn.New(ctx, vp.GetString(config.KeyServer), vp.GetDuration(config.KeyTimeout))
+			if err != nil {
+				return err
+			}
+			defer hubbleConn.Close()
+
+			return runRecord(ctx, hubbleConn, filters)
+		},
+	}
+
+	recorderFlags := pflag.NewFlagSet("recorder", pflag.ContinueOnError)
+	recorderFlags.StringVar(&fileSinkPrefix, "file-prefix", "hubble", "File prefix of the resulting pcap file")
+	recorderFlags.Uint32Var(&maxCaptureLength, "max-capture-len", 0, "Sets the maximum capture length (zero for full capture)")
+
+	recordCmd.Flags().AddFlagSet(recorderFlags)
+	recordCmd.SetUsageTemplate(template.Usage(config.ServerFlags, recorderFlags))
+
+	return recordCmd
+}
+
+func parseProto(proto string) (recorderpb.Protocol, error) {
+	switch strings.ToUpper(proto) {
+	case "TCP":
+		return recorderpb.Protocol_PROTOCOL_TCP, nil
+	case "UDP":
+		return recorderpb.Protocol_PROTOCOL_UDP, nil
+	case "ANY":
+		return recorderpb.Protocol_PROTOCOL_ANY, nil
+	default:
+		return 0, fmt.Errorf("unsupported protocol: %s", proto)
+	}
+}
+
+func parseFilters(filters []string) ([]*recorderpb.Filter, error) {
+	if len(filters) == 0 {
+		return nil, errors.New("need to specify at least one filter")
+	}
+
+	result := make([]*recorderpb.Filter, 0, len(filters))
+	for _, filter := range filters {
+		f := strings.SplitN(filter, fitlerSep, 5)
+		if len(f) != 5 {
+			return nil, fmt.Errorf("invalid 5-tuple filter: %q", f)
+		}
+
+		srcPrefix := f[0]
+		srcPort, err := strconv.ParseUint(f[1], 10, 16)
+		if err != nil {
+			return nil, fmt.Errorf("invalid source port: %s", f[1])
+		}
+		dstPrefix := f[2]
+		dstPort, err := strconv.ParseUint(f[3], 10, 16)
+		if err != nil {
+			return nil, fmt.Errorf("invalid destination port: %s", f[3])
+		}
+		proto, err := parseProto(f[4])
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, &recorderpb.Filter{
+			SourceCidr:      srcPrefix,
+			SourcePort:      uint32(srcPort),
+			DestinationCidr: dstPrefix,
+			DestinationPort: uint32(dstPort),
+			Protocol:        proto,
+		})
+	}
+
+	return result, nil
+}
+
+func fmtTime(resp *recorderpb.RecordResponse) string {
+	return resp.GetTime().AsTime().Format(time.RFC3339)
+}
+
+func fmtStats(stats *recorderpb.RecordingStatistics) string {
+	output := fmt.Sprintf("%d packets (%d bytes) written",
+		stats.GetPacketsCaptured(), stats.GetBytesCaptured())
+	if stats.GetPacketsLost() > 0 || stats.GetBytesLost() > 0 {
+		output = fmt.Sprintf("%s and %d packets (%d bytes) lost",
+			output, stats.GetPacketsLost(), stats.GetBytesLost())
+	}
+	return output
+}
+
+func runRecord(ctx context.Context, conn *grpc.ClientConn, filters []*recorderpb.Filter) error {
+	isTTY := false
+	if fileInfo, _ := os.Stdout.Stat(); (fileInfo.Mode() & os.ModeCharDevice) != 0 {
+		isTTY = true
+	}
```

Comment:
cosmetic suggestion:
```suggestion
	fileInfo, _ := os.Stdout.Stat()
	isTTY := (fileInfo.Mode() & os.ModeCharDevice) != 0
```

Also question: is this implentation portable? Would something like https://pkg.go.dev/golang.org/x/term#IsTerminal make sense since we have OSX and Windows builds?

Reply:
Hm. So we want to make sure that right now, `isTTY` is false on Windows, because it will not support the escape code we're using. Unfortunately `golang.org/x/term` does not have escape codes for the line reset (only color), so we probably would have to find a full-fledged library.

I'll do some digging, but I might end up only supporting Linux for now, while other OSes should just have `isTTY=false`, which will always work.

Reply:
> I'll do some digging, but I might end up only supporting Linux for now, while other OSes should just have isTTY=false, which will always work.

That is fine by me.

Reply:
Switched to a proper TTY check on Linux now. Non-Linux platforms will currently not use the TTY logic. We can always add that later, but I don't expect this command to be used outside of the (Linux-based) Cilium pod, as it only works in Hubble local mode.